src/peer_manager.rs

```diff
@@ -462,53 +461,72 @@ impl PeerManager {
     /// another ongoing (i.e. unapproved) candidate exists, or if the candidate is unsuitable for
     /// adding to our section, returns an error.
     pub fn expect_candidate(&mut self,
-                            candidate_name: XorName,
-                            client_auth: Authority<XorName>)
+                            old_pub_id: PublicId,
+                            target_interval: (XorName, XorName))
                             -> Result<(), RoutingError> {
-        if let Some((ongoing_name, _)) =
+        if let Some((ongoing_pub_id, candidate)) =
             self.candidates
                 .iter()
-                .find(|&(_, candidate)| !candidate.is_approved()) {
-            trace!("{:?} Rejected {} as a new candidate: still handling attempt by {}.",
+                .find(|&(_, cand)| !cand.is_approved()) {
+            trace!("{:?} Rejected {} as a new candidate: still handling attempt by {}->{}.",
                    self,
-                   candidate_name,
-                   ongoing_name);
+                   old_pub_id.name(),
+                   ongoing_pub_id.name(),
+                   candidate.fmt_new_name());
             return Err(RoutingError::AlreadyHandlingJoinRequest);
         }
-        self.routing_table
-            .should_join_our_section(&candidate_name)?;
```

Comment:
The check can't be carried out in exactly the same way as before since we now don't know what the relocated ID will be at this point - the relocating node has still to be told its target range, and this message is sent by the members of `X` who never get told what the new ID will be anyway.

But you're right about the RT function needing removed :)